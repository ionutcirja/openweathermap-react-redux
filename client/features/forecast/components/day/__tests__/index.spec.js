import React from 'react';
import { shallow } from 'enzyme';
import { Component } from '..';

describe('ForecastDay component', () => {
  const propsToRender = {
    day: 'Sunday',
    hoursList: [
      {
        hour: '18:00',
        weather: [
          {
            icon: '1.png',
            description: 'info1',
          },
        ],
        main: {
          temp: 1,
          humidity: 2,
          pressure: 3,
        },
      },
      {
        hour: '21:00',
        weather: [
          {
            icon: '2.png',
            description: 'info2',
          },
        ],
        main: {
          temp: 3,
          humidity: 5,
          pressure: 6,
        },
      },
    ],
    theme: {
      colours: {
        greyDark: 'grey-dark',
      },
    },
  };

  describe('render', () => {
    it('should render the day and a list of hours forecast', () => {
      const wrapper = shallow(<Component {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
