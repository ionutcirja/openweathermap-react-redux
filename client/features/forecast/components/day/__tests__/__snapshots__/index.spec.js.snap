// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDay component render should render the day and a list of hours forecast 1`] = `
<Day>
  <Title
    colour="grey-dark"
  >
    Sunday
  </Title>
  <ul>
    <WithTheme(Component)
      description="info1"
      hour="18:00"
      humidity={2}
      icon="1.png"
      key="18:00"
      pressure={3}
      temp={1}
    />
    <WithTheme(Component)
      description="info2"
      hour="21:00"
      humidity={5}
      icon="2.png"
      key="21:00"
      pressure={6}
      temp={3}
    />
  </ul>
</Day>
`;
